sklearn/decomposition/dict_learning.py

```diff
@@ -309,19 +309,28 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
     return code
 
 
-def _update_dict(dictionary, Y, code, verbose=False, return_r2=False,
-                 random_state=None):
+def _proj_l2(v):
+    """Projects v unto l2 unit ball in-place.
+    """
+    vv = np.dot(v, v)
+    if vv > 1.:
+        v /= sqrt(vv)
+    return v
+
+
+def _update_dict(dictionary, B, A, verbose=False, return_r2=False,
+                 random_state=None, online=False):
     """Update the dense dictionary factor in place.
 
     Parameters
     ----------
     dictionary : array of shape (n_features, n_components)
         Value of the dictionary at the previous iteration.
 
-    Y : array of shape (n_features, n_samples)
+    B : array of shape (n_features, n_components)
         Data matrix.
 
-    code : array of shape (n_components, n_samples)
+    A : array of shape (n_components, n_components)
```

Comment:
I added a more detailed comment below after thinking about it. No, the normalization makes sense to me. It just seems like the old computation does not correspond to what is in the paper.